shifthelper/checks.py

```diff
@@ -340,18 +343,147 @@ class TriggerRateLowForTenMinutes(FactIntervalCheck):
 
     def inner_check(self):
         current_trigger_rate = sfc.trigger_rate().trigger_rate.value
+        _is_data_taking = is_data_taking()
         self._append_to_history(current_trigger_rate)
         self._remove_old_entries()
 
         df = pd.DataFrame(self.history)
-        log.debug('TriggerRateLowForTenMinutes: trigger_rate_history:{0}'.format(df))
-        if not df.empty and (df.rate < 1).all():
-            return 'Trigger rate < 1/s for 10 minutes'
+        log.debug(
+            'TriggerRateLowForTenMinutes: (rate < 1).all:{0} data_taking:{1}'.format(
+                (df.rate < 1).all(), _is_data_taking))
+        if _is_data_taking and not df.empty and (df.rate < 1).all():
+            return 'Trigger rate < 1/s for 10 minutes, while data taking'
 
     def _append_to_history(self, rate):
-        self.history = self.history.append([{'timestamp':datetime.utcnow(), 'rate':rate}])
+        self.history = self.history.append([{'timestamp': datetime.utcnow(), 'rate': rate}])
 
     def _remove_old_entries(self):
         now = datetime.utcnow()
         self.history = self.history[(now - self.history.timestamp) < timedelta(minutes=10)]
 
+
+class IsUserAwakeBeforeShutdown(FactIntervalCheck):
+    '''
+    This Check should find out if the user is actually awake some time before
+    the scheduled shutdown.
+
+    In order to find out, if the shifter is awake, we need to find out:
+      * Is it 20minutes (or less) before shutdown?
+      * Who is the current shifter (username)?
+      * Is she awake, i.e. did he press the "I am awake button" recently?
+    '''
+
+    def inner_check(self):
+        try:
+            shutdown_time = get_next_shutdown().fStart
+        except IndexError:
+            return
+
+        earliest_awake_time = shutdown_time - timedelta(minutes=30)
+        current_shifter = whoisonshift()
+
+        user_since = requests.get('https://ihp-pc41.ethz.ch/iAmAwake').json()
```

Comment:
Not sure. I guess, yes. What should the reaction be, in case it cannot find the website? Try a couple of seconds .. up to 1 minute and then call, right?